### PR TITLE
STM32F03: Extended USART functions (data/pin inversion, half duplex)

### PR DESCRIPTION
**Pull request by cemizm**

### CODE REVIEW

File: lib/stm32/common/usart_common_f03.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
+/** @addtogroup usart_file
+
+ @author @htmlonly &copy; @endhtmlonly 2016 Cem Basoglu <cem.basoglu@web.de>
+
+ */
+
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2016 Cem Basoglu <cem.basoglu@web.de
```
Comment (karlp) [inline lib/stm32/common/usart_common_f03.c]: missing a closing > here

File: lib/stm32/common/usart_common_f03.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+#include <libopencm3/stm32/usart.h>
+
+/*---------------------------------------------------------------------------*/
+/** @brief USART enable data inversion
+
+ Logical data from the data register are send/received in negative/inverse logic. (1=L, 0=H). The parity bit is also inverted.
+
+ @This bit field can only be written when the USART is disabled.
+
+ @param[in] usart unsigned 32 bit. USART block register address base @ref
```
Comment (karlp) [inline lib/stm32/common/usart_common_f03.c]: no need for "unsigned 32 bit" that's conveyed alreay by the function signatures.

File: lib/stm32/common/usart_common_f03.c [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+ */
+
+/**@{*/
+
+#include <libopencm3/stm32/usart.h>
+
+/*---------------------------------------------------------------------------*/
+/** @brief USART enable data inversion
+
+ Logical data from the data register are send/received in negative/inverse logic. (1=L, 0=H). The parity bit is also inverted.
```
Comment (karlp) [inline lib/stm32/common/usart_common_f03.c]: long lines

File: lib/stm32/f3/Makefile [hunk 4]
```diff
@@ -38,8 +38,8 @@ ARFLAGS		= rcs
 
 OBJS		= rcc.o adc.o i2c.o usart.o dma.o flash.o
 
-OBJS            += gpio_common_all.o gpio_common_f0234.o \
-		   dac_common_all.o usart_common_all.o crc_common_all.o\
+OBJS            += gpio_common_all.o gpio_common_f0234.o usart_common_f03.o \
+		   dac_common_all.o usart_common_all.o crc_common_all.o crc_common_f03.o \
```
Comment (karlp) [inline lib/stm32/f3/Makefile]: crc common shouldn't be part of this commit.

**Comment by karlp** [comment]
I like it, but again, this is the "new" usart peripheral used on f0, f3, l0, l4, f7.  Please just call the file usart_common_v2.

**Comment by cemizm** [comment]
Renamed to common v2 and implemented convenience functions for receive timeout functionality.

**Comment by fenugrec** [comment]
@cemizm you will probably need to rebase / squash those commits into one

Otherwise, looks fine, and I verified the "USART_RTOR_BLEN1_MASK" change will break nothing in the examples repo

**Comment by karlp** [comment]
merged after rebase and some whitespace tweaks in 6b5150a thanks for this!
